nats-base-client/headers.ts

```diff
@@ -182,24 +182,24 @@ export class MsgHdrsImpl implements MsgHdrs {
     return k.trim();
   }
 
-  get(k: string): string {
-    const key = MsgHdrsImpl.canonicalMIMEHeaderKey(k);
+  get(k: string, mime = true): string {
+    const key = mime ? MsgHdrsImpl.canonicalMIMEHeaderKey(k) : k;
     const a = this.headers.get(key);
     return a ? a[0] : "";
   }
 
-  has(k: string): boolean {
-    return this.get(k) !== "";
+  has(k: string, mime = true): boolean {
+    return this.get(k, mime) !== "";
   }
 
-  set(k: string, v: string): void {
-    const key = MsgHdrsImpl.canonicalMIMEHeaderKey(k);
+  set(k: string, v: string, mime = true): void {
+    const key = mime ? MsgHdrsImpl.canonicalMIMEHeaderKey(k) : k;
```

Comment:
I would avoid transforming the header key here and just take it as is, then only when doing a get have an option of whether the library is expecting the case to preserved or not.